### PR TITLE
react-map-gl - Added missing types

### PR DESCRIPTION
**Pull request by Arman92**
Fixed missing types according to: https://github.com/uber/react-map-gl/blob/5.1-release/examples/viewport-animation/src/app.js
FlyToInterpolator class was missing it's constructor props and transitionDuration property couldn't have 'auto' value

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/uber/react-map-gl/blob/3e35c9b78430cdc19f1072806ff2545966ac4542/examples/viewport-animation/src/app.js#L30

### CODE REVIEW

**Comment by oturpe** [comment]
The main content looks good, thank you @Arman92 !

Both I and the test automation have the complaint that some autoformatter has been run on top of the codebase. Could you revert any autoformatting and only include changes related to the new type, using the style that is already used?

File: types/react-map-gl/index.d.ts
Comment (fnberta) [inline types/react-map-gl/index.d.ts]: Can we make this an interface instead? The rest of the types use interfaces when possible.

**Review by fnberta — CHANGES_REQUESTED** [review]
Thanks for the PR, looks good, only left one minor comment. Regarding the formatting, I actually welcome the changes. But maybe have them as a separate PR, that makes it clearer what was changed.

File: types/react-map-gl/index.d.ts
Comment (Arman92) [reply types/react-map-gl/index.d.ts]: Sure, Actually the builds were failing because I used 'type' instead of 'interface', just fixed it in a new commit.

**Comment by oturpe** [comment]
Just to make it clear, I actually do not have any position regarding how the codebase should be formatted. The current style does for me as well as any other would. My concern was related to mixing type updates and reformatting in a single MR.

**Comment by Arman92** [comment]
@oturpe Sure, you are absolutely right. I didn't notice that my prettier tool did a reformat while I saved files, I do agree with you; code reformatting should be in a different PR.

**Comment by fnberta** [comment]
There are still formatting changes in the tests file. Could you revert these as well and put in a separate PR? Thanks!

**Review by fnberta — APPROVED** [review]

**Comment by Arman92** [comment]
This PR is still waiting for someone to kindly apply the merge!

**Comment by Arman92** [comment]
@oturpe, @fnberta This PR is still pending, maybe it's because it's still labeled 'Revision Needed'.

Can you remove this label? Or should I ask a maintainer to do this?

**Comment by fnberta** [comment]
Hmm I dont think I can do anything besides approving which I already did.

**Pull request merged**
